AWSAppSyncClient/AWSAppSyncCache.swift

```diff
@@ -10,7 +10,6 @@ import Foundation
 public enum AWSAppSyncQueriesCacheError: Error {
     case invalidRecordEncoding(record: String)
     case invalidRecordShape(object: Any)
-    case invalidRecordValue(value: Any)
```

Comment:
can this be added back in? changes to the public interfaces will break existing customers or we will have to do a major version bump. Unless absolutely necessary, I think we should keep the enum case even if it's no longer used and in the future remove it completely in a major version bump. The incremental release will let existing developers just upgrade to a minor version to take the new behavior but not have to change the code that uses this enum

Reply:
I can absolutely put it back, and totally understand the reasoning. Would you like me to mark that case as `@available(*, deprecated)`, since it's no longer used or thrown from anywhere?

Reply:
sure that makes sense, thanks! Was thinking something like this:
```
@available(*, deprecated, message: "No longer being used and will be removed in a future revision.")
```

Reply:
Done and done! 😊